core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java

```diff
@@ -430,6 +437,47 @@ String generateSep10Jwt(ChallengeTransaction challenge, String clientDomain, Str
     debug("jwtToken:", sep10Jwt);
     return jwtService.encode(sep10Jwt);
   }
+
+  /**
+   * Extracts the home domain from a Stellar SEP-10 challenge transaction represented in XDR format.
+   *
+   * @param challengeXdr SEP-0010 transaction challenge transaction in base64.
+   * @param network The network to connect to for verifying and retrieving.
+   * @return The extracted home domain from the Manage Data operation within the SEP-10 challenge
+   *     transaction.
+   * @throws IOException If read XDR string fails, the exception will be thrown.
+   * @throws InvalidSep10ChallengeException If the SEP-0010 validation fails, the exception will be
```

Comment:
This isn't really SEP-10 validation failing, rather it is the SEP-10 challenge transaction being malformed.